cmd/command.go

```diff
@@ -0,0 +1,444 @@
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/sirupsen/logrus"
+	"github.com/taosdata/taoskeeper/db"
+	"github.com/taosdata/taoskeeper/infrastructure/config"
+	"github.com/taosdata/taoskeeper/infrastructure/log"
+	"github.com/taosdata/taoskeeper/util"
+	"github.com/taosdata/taoskeeper/util/pool"
+)
+
+var logger = log.GetLogger("command")
+
+var MAX_SQL_LEN = 1000000
+
+type Command struct {
+	fromTime time.Time
+	client   *http.Client
+	conn     *db.Connector
+	username string
+	password string
+	url      *url.URL
+}
+
+func NewCommand(conf *config.Config) *Command {
+	client := &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).DialContext,
+			IdleConnTimeout:       90 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
+			DisableCompression:    true,
+		},
+	}
+
+	conn, err := db.NewConnectorWithDb(conf.TDengine.Username, conf.TDengine.Password, conf.TDengine.Host, conf.TDengine.Port, conf.Metrics.Database)
+	if err != nil {
+		logger.Error("## init db connect error", "error", err)
+		panic(err)
+	}
+
+	imp := &Command{
+		client:   client,
+		conn:     conn,
+		username: conf.TDengine.Username,
+		password: conf.TDengine.Password,
+		url: &url.URL{
+			Scheme:   "http",
+			Host:     fmt.Sprintf("%s:%d", conf.TDengine.Host, conf.TDengine.Port),
+			Path:     "/influxdb/v1/write",
+			RawQuery: fmt.Sprintf("db=%s&precision=ms", conf.Metrics.Database),
+		},
+	}
+	return imp
+}
+
+func (cmd *Command) Process(conf *config.Config) {
+	if len(conf.Transfer) > 0 && conf.Transfer != "old_taosd_metric" {
+		logger.Errorf("transfer only support old_taosd_metric")
+		return
+	}
+
+	if conf.Transfer == "old_taosd_metric" {
+		cmd.ProcessTransfer(conf)
+		return
+	}
+
+	if len(conf.Drop) > 0 && conf.Drop != "old_taosd_metric_stables" {
+		logger.Errorf("drop only support old_taosd_metric_stables")
+		return
+	}
+
+	if conf.Drop == "old_taosd_metric_stables" {
+		cmd.ProcessDrop(conf)
```

Comment:
If the transfer fails, the drop will still be executed

Reply:
add check